js/utils/getBitcoinPrice.js

```diff
@@ -4,143 +4,164 @@ var __ = require('underscore'),
-      });
-
-  var callCoindesk = function ()
-  {*/
-      $.ajax({
-        method: "GET",
-        dataType: "json",
-        url: "https://api.coindesk.com/v1/bpi/currentprice/" + currency + ".json"
-      })
-          .done(function (response)
-          {
-            //console.log("coinDesk: " + response.bpi[currency]['rate']);
-            if($.isNumeric(response.bpi[currency].rate)) {
-              btPrices.push(response.bpi[currency].rate);
-            }
-          })
-          .fail(function (jqXHR, textStatus, errorThrown)
-          {
-            //console.log("coinDesk request failed:");
-            //console.log(jqXHR);
-            //console.log(textStatus);
-            //console.log(errorThrown);
-          })
-          .always(function ()
-          {
-            callBlockchain();
-          });
-    //};
+        window.btcAverages = {};
+        var btPrices = [];
+        var btcAverages = {rates: {}};
 
+        var callBlockchain = function () {
+            $.ajax({
+                method: "GET",
+                url: "https://blockchain.info/ticker"
+            })
+                .done(function (response) {
+                    var BlockchainCurrencies = {};
+                    for (var bcCurrency in response) {
+                        if (response.hasOwnProperty(bcCurrency)) {
+                            BlockchainCurrencies[bcCurrency] = response[bcCurrency]['15m'];
+                        }
+                    }
+                    btPrices.push(BlockchainCurrencies);
+                })
+                .fail(function (jqXHR, textStatus, errorThrown) {
+                    console.log("Blockchain request failed: ");
+                    console.log(jqXHR);
+                    console.log(textStatus);
+                    console.log(errorThrown);
+                })
+                .always(function () {
+                    callCoinkite();
+                });
+        };
 
-    var callBlockchain = function ()
-    {
-      $.ajax({
-        method: "GET",
-        url: "https://blockchain.info/ticker"
-      })
-          .done(function (response)
-          {
-            //console.log("blockChain: " + response[currency]['15m']);
-            if($.isNumeric(response[currency]['15m'])) {
-              btPrices.push(response[currency]['15m']);
-            }
-          })
-          .fail(function (jqXHR, textStatus, errorThrown)
-          {
-            //console.log("blockChain request failed: ");
-            //console.log(jqXHR);
-            //console.log(textStatus);
-            //console.log(errorThrown);
-          })
-          .always(function ()
-          {
-            callCoinKite();
-          });
-    };
+        var callCoinkite = function () {
+            $.ajax({
+                method: "GET",
+                url: "https://api.coinkite.com/public/rates"
+            })
+                .done(function (response) {
+                    var CoinkiteCurrencies = {};
```

Comment:
It's because .BTC holds all rates for every different currency as well as a lot of unneeded information. If we pushed response.rates.BTC it would push all of the following that we don't need in a different format than the other api's so calling the averages would be different for this one...
code: "AUD",
pair: "BLK/AUD",
pretty: "$ 0.04",
pretty_rev: "Ѣ 24.6687",
rate: 0.04053727,
sampled_at: "2015-09-28T11:38:04.407196Z"